libraries/commerce/favorites/streams/index.js

```diff
@@ -47,3 +49,20 @@ export const favoritesWillRemoveItem$ = main$
   .filter(({ action }) => action.type === REQUEST_REMOVE_FAVORITES && !action.silent);
 
 export const favoritesSyncIdle$ = main$.filter(({ action }) => action.type === IDLE_SYNC_FAVORITES);
+
+export const addRemoveFavorites$ = main$.filter(({ action }) => (
+  action.type === REQUEST_ADD_FAVORITES || action.type === REQUEST_REMOVE_FAVORITES
```

Comment:
The `REQUEST_REMOVE_FAVORITES` action could be fired with a `silent` attribute, which is used to remove a favorite from the redux store only (mostly when adding fails). In this case the attribute is ignored, which causes a remove request to be sent to the backend, which should not happen, because the favorite was not really added in that case.

Reply:
good catch!